src/components/Select2.js

```diff
@@ -46,6 +52,23 @@ export default class Select2 extends Component {
     this.props.events.forEach(event => {
       this.el.on(event[0], this.props[event[1]]);
     });
+    const { defaultValue, value } = this.props;
+    if (!defaultValue && this.el.val() !== value) {
+      this.setValue(value);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.el && nextProps.value !== this.props.value) {
+      this.setValue(nextProps.value);
+    }
+  }
+
+  setValue(value) {
+    const elVal = this.el.val();
+    if (!shallowEqual(elVal, value)) {
```

Comment:
I add shallowEqual here. It's nessasary for multiple value, when we need to compare 2 arrays. for example [1,2] !== [1,2] but shallowEqual([1,2], [1,2]) === true.